pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type DropReason string
+
+type Phase string
+
+const (
+	INITIAL Phase = "INITIAL"
+	RUNNING Phase = "RUNNING"
+	SUCCESS Phase = "SUCCESS"
+	TIMEOUT Phase = "TIMEOUT"
+	ERROR   Phase = "ERROR"
+)
+
+type Component string
+
+const (
+	SPOOFGUARD    Component = "SPOOFGUARD"
+	LB            Component = "LB"
+	ROUTING       Component = "ROUTING"
+	NETWORKPOLICY Component = "NETWORKPOLICY"
+	FORWARDING    Component = "FORWARDING"
+)
+
+type Action string
+
+const (
+	DELIVERED Action = "DELIVERED"
+	RECEIVED  Action = "RECEIVED"
+	FORWARDED Action = "FORWARDED"
+	DROPPED   Action = "DROPPED"
+)
+
+type PacketTransportType string
+
+const (
+	UNICAST PacketTransportType = "UNICAST"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Traceflow struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Source      `json:"source"`
+	Destination `json:"destination"`
+	RoundID     string `json:"roundID,omitempty"`
```

Comment:
I remember we talked about a better name for this one? At least add some comments to explain what it is.
If it is not required by Antrea itself but by some clients, we might remove it, and the clients can use annotation for such client IDs.

Reply:
@jianjuns Please comment CRD design in PR https://github.com/vmware-tanzu/antrea/pull/660 , this PR just cherry-pick Zhecheng's commit and adapt the code to latest CRD design.

Reply:
Please explicitly note this in the PR description for such things. Otherwise how reviewers can understand.
@lzhecheng

Reply:
Updated PR description for this note.

Reply:
Move to:
https://github.com/vmware-tanzu/antrea/pull/660#discussion_r437849590